src/main/java/com/google/api/generator/engine/writer/JavaWriterVisitor.java

```diff
@@ -120,4 +134,97 @@ public void visit(AssignmentExpr assignmentExpr) {
   private void space() {
     buffer.append(SPACE);
   }
+
+  /** =============================== STATEMENT =============================== */
+  public String visit(LineComment linecomment){
+      // Split comments by new line and add `//` to each line.
+      String[] sourceStrings = linecomment.comment().split("\\r?\\n");
+      for (int i = 0; i < sourceStrings.length; i++) {
+        sourceStrings[i] = "// " + sourceStrings[i];
+      }
+      String formattedSource =format(String.join("\n", sourceStrings));
+      return formattedSource;
+  }
+
+  public String visit(BlockComment blockComment){
+    // Split comments by new line and embrace the comment block with `/** */`.
+    String sourceString = blockComment.comment();
+    String formattedSource = format("/** " + sourceString + " */");
+    return formattedSource;
+  }
+
+  public String visit(JavaDocComment javaDocComment) {
+    StringBuilder formattedComment = new StringBuilder("/**\n");
+    List<String> commentList = new ArrayList<>(javaDocComment.comments());
+    Map<String, String> paramList = new HashMap<>(javaDocComment.params());
+    List<String> sampleCodeList = new ArrayList<>(javaDocComment.sampleCode());
+    List<String> html_p = new ArrayList<>(javaDocComment.html_p());
+    List<List<String>> html_ol = new ArrayList<>(javaDocComment.html_ol());
+    List<List<String>> html_ul = new ArrayList<>(javaDocComment.html_ol());
+    List<COMMENT_TYPE> orderList = new ArrayList<>(JavaDocComment.commentList);
+
+    for(COMMENT_TYPE type : orderList){
```

Comment:
Will look at this in the next pass (please see JavaDocComment comments above).

Reply:
👍 sounds good, thank you